Objects/bytesobject.c

```diff
@@ -3744,12 +3744,49 @@ _PyBytes_Resize(PyObject **pv, Py_ssize_t newsize)
     return 0;
 }
 
+/* The following initializes `nullstring' and `characters' to avoid races
+   caused by lazily computing them.  Effectively, we're pessimizing any
+   programs that don't use empty or single character bytestrings, but it's a
+   one time cost and the upside is thread safety.
+ */
+int
+_PyBytes_Init(void)
+{
+    /* Inline PyObject_NewVar */
+    size_t size;
+    PyBytesObject *op;
+    char str[] = "\0\0";
+    int i;
+
+    size = 0;
+    // Allocate the nullstring.
+    op = (PyBytesObject *)PyObject_Malloc(PyBytesObject_SIZE + size);
+    if (op == NULL)
+       return 0;
+    (void)PyObject_INIT_VAR(op, &PyBytes_Type, size);
+    Py_MEMCPY(op->ob_sval, str, size+1);
+    op->ob_shash = -1;
+    nullstring = op;
+
+    size = 1;
+    // Now we allocate all the 1 byte character strings.
+    op = (PyBytesObject *)PyObject_Calloc(UCHAR_MAX, PyBytesObject_SIZE + size);
```

Comment:
It seemed like a good idea to use calloc to get an array of these objects, but I suspect I'm getting the reference counting wrong as a result.